Automattic/jetpack
# Jetpack At-a-glance: Replace Backup upgrade banner with a coupon redeem CTA when a partner coupon is detected

#### Changes proposed in this Pull Request:

* Replace the Backup upgrade banner on the "At a glance" page with a coupon redeem CTA when a partner coupon is detected.

#### Jetpack product discussion
pdpAdu-1A-p2

#### Does this pull request change what data or activity we track or use?

Yes, in Tracks:
- `jetpack_wpa_click`: Now can also be triggered with 2 different props:
    - `type`: "redeem_partner_coupon"
    - `coupon_preset`: The unique slug that determines which preset a partner coupon was generated from.
- `jetpack_wpa_aag_redeem_partner_coupon_button_view`: Has been added with 2 props:
    - `feature`: "backups" (mirroring the `jetpack_wpa_aag_upgrade_button_view` event)
    - `coupon_preset`: The unique slug that determines which preset a partner coupon was generated from.

Examples: 2b966-pb/#plain

#### Testing instructions:

- Make sure Jetpack is activated and connected, has Jetpack Free as its plan, and has no other products.
- Visit `/wp-admin/admin.php?page=jetpack#/dashboard` and confirm the Backup feature section shows what is visible in screenshot 1 below.
- Visit `/wp-admin/?jetpack-partner-coupon=PARTNER_COUPON_CODE` to store a partner coupon code. Please reach out to team Avalon to get a coupon code.
- Visit `/wp-admin/admin.php?page=jetpack#/dashboard` and confirm the Backup feature section shows what is visible in screenshot 2 below. Clicking on the Redeem button should redirect you to checkout with your partner coupon applied.
- Purchase a product to your test site that provides backups.
- Visit `/wp-admin/admin.php?page=jetpack#/dashboard` and confirm the Backup feature section shows what is visible in screenshot 3 below.

#### Final results:

Screenshot 1) No backup product, no partner coupon:

![Screen Shot 2022-02-01 at 19 08 35](https://user-images.githubusercontent.com/22746396/152016113-1da5365c-3f54-40c9-b079-47e97767bd40.png)

Screenshot 2) No backup product, partner coupon available:

![Screen Shot 2022-02-01 at 19 03 52](https://user-images.githubusercontent.com/22746396/152016107-bf474b8b-2970-44f5-b11a-b5f997bb613c.png)

Screenshot 3) Backup product active, partner coupon status is irrelevant:

![Screen Shot 2022-02-01 at 18 57 18](https://user-images.githubusercontent.com/22746396/152016100-6fe4a8f5-4ac4-482e-9501-8ed7f216f592.png)
